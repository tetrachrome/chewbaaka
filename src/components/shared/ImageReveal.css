/**
 * ImageReveal.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Updated  : Jul 05, 2020
 * Created  : Jul 05, 2020
 */

div.ImageRevealComponentContainerDiv {
  width: 120px;
  height: 120px;
}

div.ImageRevealComponentContentDiv {
  width: 120px;
  height: 120px;
}
