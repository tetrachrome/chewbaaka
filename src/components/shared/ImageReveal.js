/**
 * ImageReveal.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 04, 2020
 * Updated  : Jul 05, 2020
 */

/**
 * ImageReveal
 *
 * ImageReveal is a component that initially shows
 * a cover image specified by the user, and when
 * hovered, hides the cover image and shows another
 * image specified by the user.
 *
 * Props:
 *
 *  - `coverImage`: The cover image.
 *
 *  - `contentImage`: The content image to show upon hovered.
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Image, Reveal } from 'semantic-ui-react'

import './ImageReveal.css'

class ImageReveal extends React.Component {

  render() {
    const size = this.props.size ? this.props.size : "small";

    return (
      <Reveal animated={this.animation()} data-testid="ImageRevealComponentTestId">
        <Reveal.Content visible data-testid="ImageRevealComponentVisiblePartTestId">
          <div className="ImageRevealComponentContainerDiv">
            <Image src={this.props.coverImage} />
          </div>
        </Reveal.Content>
        <Reveal.Content hidden data-testid="ImageRevealComponentHiddenPartTestId">
          <div className="ImageRevealComponentContentDiv">
            <Image src={this.props.contentImage} size={size} />
          </div>
        </Reveal.Content>
      </Reveal>
    )
  }

  animation() {
    const supportedAnimations = [
      "fade",
      "small fade",
      "move",
      "move right",
      "move up",
      "move down",
      "rotate",
      "rotate left"
    ];

    let idx = supportedAnimations.indexOf(this.props.animation);

    if ( idx === -1 ) {
      idx = Math.floor(Math.random() * supportedAnimations.length);
    }

    return supportedAnimations[idx];
  }
}

export default ImageReveal;
