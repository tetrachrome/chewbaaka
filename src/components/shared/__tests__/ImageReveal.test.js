/**
 * ImageReveal.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 04, 2020
 * Updated  : Jul 08, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import ImageReveal from '../ImageReveal';

import coverImage from '../assets/cheetah-conservation-fund-logo.jpg'
import contentImage from '../assets/cheetah-conservation-fund-30-logo.svg'

test('renders ImageReveal compoent', () => {
  render(<ImageReveal coverImage={coverImage} contentImage={contentImage} size="small"/>);

  const imageRevealComponent = screen.getByTestId('ImageRevealComponentTestId');
  expect(imageRevealComponent).toBeInTheDocument();

  const imageRevealComponentVisiblePart = screen.getByTestId('ImageRevealComponentVisiblePartTestId');
  expect(imageRevealComponentVisiblePart).toBeInTheDocument();

  const imageRevealComponentHiddenPart = screen.getByTestId('ImageRevealComponentHiddenPartTestId');
  expect(imageRevealComponentHiddenPart).toBeInTheDocument();

  // TODO: test component props.
});

test('ImageReveal component snapshot', () => {
  const tree = renderer
    .create(
      <ImageReveal
        coverImage={coverImage}
        contentImage={contentImage}
        size="small"
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
