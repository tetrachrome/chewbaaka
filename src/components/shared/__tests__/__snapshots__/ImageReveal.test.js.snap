// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageReveal component snapshot 1`] = `
<div
  className="ui fade reveal"
  data-testid="ImageRevealComponentTestId"
>
  <div
    className="ui visible content"
    data-testid="ImageRevealComponentVisiblePartTestId"
  >
    <div
      className="ImageRevealComponentContainerDiv"
    >
      <img
        className="ui image"
        src="cheetah-conservation-fund-logo.jpg"
      />
    </div>
  </div>
  <div
    className="ui hidden content"
    data-testid="ImageRevealComponentHiddenPartTestId"
  >
    <div
      className="ImageRevealComponentContentDiv"
    >
      <img
        className="ui image"
        src="cheetah-conservation-fund-30-logo.svg"
      />
    </div>
  </div>
</div>
`;
